pandas/core/indexes/base.py

```diff
@@ -2584,6 +2584,11 @@ def get_indexer(self, target, method=None, limit=None, tolerance=None):
         if tolerance is not None:
             tolerance = self._convert_tolerance(tolerance)
 
+        if target.is_boolean() and self.is_numeric():
+            warnings.warn('Index is numeric and indexer is boolean, '
+                          'possible unintended conversion from boolean '
+                          'to numeric', UserWarning, stacklevel=1)
```

Comment:
I think we can rephrase the prose here.  Proper grammar and slightly more formality in the writing would be good here.

Reply:
'Passing a boolean indexer to an Index containing numeric indices may cause the indexer to behave as being of numeric type'

@gfyoung Thanks for the feedback. Would the above work? Any pointers would be welcome.